src/backend/distributed/commands/create_distributed_table.c

```diff
@@ -556,11 +556,15 @@ CreateDistributedTable(Oid relationId, Var *distributionColumn, char distributio
 	{
 		List *partitionList = PartitionList(relationId);
 		Oid partitionRelationId = InvalidOid;
+		Oid namespaceId = get_rel_namespace(relationId);
+		char *schemaName = get_namespace_name(namespaceId);
+		char *relationName = get_rel_name(relationId);
+		char *parentRelationName = quote_qualified_identifier(schemaName, relationName);
 		foreach_oid(partitionRelationId, partitionList)
 		{
 			CreateDistributedTable(partitionRelationId, distributionColumn,
 								   distributionMethod, shardCount, false,
-								   colocateWithTableName, viaDeprecatedAPI);
```

Comment:
What happens if the parent relation name is "none"?

Reply:
Hmm, I thought these are keywords when I was reading the code
    if (pg_strncasecmp(colocateWithTableName, "**default**", NAMEDATALEN) != 0)

    pg_strncasecmp(colocateWithTableName, "**none**", NAMEDATALEN) == 0;

Reply:
I got it now, you meant for the parent relation :)

Reply:
Parent name can be either 'none' or 'schema.none', in both the cases we resolve it 'schema.none', using the parent Oid, which shouldn't match.

pg_strncasecmp(colocateWithTableName, "default", NAMEDATALEN) != 0)
pg_strncasecmp(colocateWithTableName, "none", NAMEDATALEN) == 0